### PR TITLE
Example: Improving Escrow Program

### PR DESCRIPTION
**Pull request by ironaddicteddog**
This is a PR for improved escrow program example. There are few changes including:

- Add Readme
- Instead of letting initializer create a token account to be reset to a PDA authority, improved example uses a token account `Vault` that has both a PDA key and a PDA authority. (This idea is actually came from @xjcaa)
- Implement `close_account` in `anchor_spl`.

Also I am working on an [explainer](https://hackmd.io/@ironaddicteddog/anchor_example_escrow) on this example too. Feedbacks are welcome!

### CODE REVIEW

**Comment by armaniferrante** [comment]
There's a bunch of conflicts on this. Sorry about that!

**Comment by ironaddicteddog** [comment]
@armaniferrante Let me fix it 💪

File: tests/escrow/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,5396 @@
+{
```
Comment (armaniferrante) [inline tests/escrow/package-lock.json:1]: Please remove the package-lock. The repo is not committing them for the examples.

Comment (ironaddicteddog) [reply tests/escrow/package-lock.json]: Fixed

**Comment by Henry-E** [comment]
Seems like this is still blocked. 🤷 it's a close for now

**Pull request closed without merging**